Merging new branch annotations_dump (which has xml to json conversion script) to master

KeshavBalaji opened the pull request:
Adding standalone Python script that parses xml file of annotations and writes to corresponding json file

KeshavBalaji committed:
Add files via upload

Adding standalone Python script that parses xml file of annotations and writes to corresponding json file

iross commented on xml_to_json.py:
```diff
@@ -0,0 +1,86 @@
+"""
+
+import os
+import glob
```
Unused  -- can be removed

iross commented on xml_to_json.py:
```diff
@@ -0,0 +1,86 @@
+the data to a corresponding json file
+"""
+
+import os
```
Unused -- can be removed

iross commented on xml_to_json.py:
```diff
@@ -0,0 +1,86 @@
+import sys
+from pydantic import BaseModel, Field
+from typing import List
+from numpy import arange
```
Unused -- can be removed

iross commented on xml_to_json.py:
```diff
@@ -0,0 +1,86 @@
+        """ Get the intersection-over-union of two rectangles """
+        return self.intersection(other) / self.union(other)
+
+def parse_xml() -> list[AnnotationBounds]:
```
Add a docstring to explain what inputs and outputs this method expects and returns

iross commented on xml_to_json.py:
```diff
@@ -0,0 +1,86 @@
+
+    return annotations, xml_data.name[:-4]
+
+def _cosmos_obj_from_manual_obj(page_num, obj):
```
Typically an underscore in a method name indicates that it is intended for private use (e.g. within a class) -- see https://realpython.com/python-double-underscore/

iross commented on xml_to_json.py:
```diff
@@ -0,0 +1,86 @@
+            bounding_box= bounds
+        )
+
+def create_json(xml_annotations, filename):
```
Add a docstring here too

iross commented on xml_to_json.py:
```diff
@@ -0,0 +1,86 @@
+File: xml_to_json.py
+Author: Keshav Balaji
+Description: Standalone Python script to parse an xml file of annotations and write
+the data to a corresponding json file
```
A usage example is useful here to show what it expects as inputs defined at the command line

iross commented on xml_to_json.py:
```diff
@@ -0,0 +1,86 @@
+from numpy import arange
+
+DEFAULT_REGION_TYPES = ["Figure", "Equation", "Table"]
+# BASE_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__),'../..'))
```
Let's go ahead and delete this kind of code remnant.

iross reviewed (commented):
Looks good! I left a few comments. I'll leave this open for now -- let's also put the  XML reorganization script on this branch and repo as well.

KeshavBalaji committed:
Delete xml_to_json.py

going to soon add the same file with comments/doctags and without unnecessary lines

KeshavBalaji committed:
Adding standalone Python scripts

Adding organize_ground_truths.py and xml_to_json.py to organize xml files and pdfs into directories, and create json files for each corresponding xml files

KeshavBalaji commented:
Merging annotations_dump branch with updated xml_to_json.py and organize_ground_truths.py to master
